tests/unit/nupic/algorithms/knn_classifier_test.py

```diff
@@ -30,6 +30,67 @@
 class KNNClassifierTest(unittest.TestCase):
 
 
+  def testDistanceMetrics(self):
+    classifier = KNNClassifier(distanceMethod="norm", distanceNorm=2.0)
+  
+    dimensionality = 40
+    protoA = np.array([0, 1, 3, 7, 11], dtype=np.int32)
+    protoB = np.array([20, 28, 30], dtype=np.int32)
+    
+    classifier.learn(protoA, 0, isSparse=dimensionality)
+    classifier.learn(protoB, 0, isSparse=dimensionality)
+    
+    input = np.zeros(dimensionality)
+    input[:4] = 1.0
+    
+    # Test l2 norm metric
+    _, _, dist, _ = classifier.infer(input)
```

Comment:
Nice test cases. Please also include edge cases, i.e. when distance should be 0 or the largest possible.